tools/AutomationTestFramework/ManageScripts/DeliverModuleToAutomationAccount.ps1

```diff
@@ -0,0 +1,187 @@
+Container name in the Azure Storage Account
+.NOTES
+This is the first step of the two steps upload to Azure Automation Account action
+#>
+function UploadModuleToContainer (
+     [string] $modulePath
+    ,[string] $moduleName
+    ,[string] $subscriptionName
+    ,[string] $storageAccountName
+    ,[string] $resourceGroupName
+    ,[string] $containerName) {
+
+    $subscriptionName, $resourceGroupName, $storageAccountName,  $containerName = DefaultIfNotSpecifiedSA $subscriptionName $resourceGroupName $storageAccountName  $containerName
+    Write-Host "Getting Azure storage account key..." -ForegroundColor Green
+
+    $storageAccountKey = Get-AzureRmStorageAccountKey -StorageAccountName $storageAccountName -ResourceGroupName $resourceGroupName -Verbose -ErrorAction Stop
+    $key1 = $storageAccountKey.Value[0];
+    Write-Host "Creating Azure storage account context..." -ForegroundColor Green
+    $ctx = New-AzureStorageContext $storageAccountName -StorageAccountKey $key1 -ErrorAction Stop
+    Write-Host "Uploading module to Azure storage container..." -ForegroundColor Green
+    $blobName = "$moduleName.zip"
+    $null = Set-AzureStorageBlobContent -Container $containerName  -File "$modulePath\$blobName" -Blob $blobName -Context $ctx -Force -ErrorAction Stop
+    $now = Get-Date;
+    Write-Host "Creating Azure storage blob SAS URL..." -ForegroundColor Green
+    $sasTokenURL = New-AzureStorageBlobSASToken -Container $containerName -Blob $blobName -Context $ctx -Permission rwd -StartTime $now.AddHours(-1) -ExpiryTime $now.AddHours(1) -FullUri -ErrorAction Stop
+    return $sasTokenURL
+}
+
+<#
+.SYNOPSIS
+Uploads module from Azure Storage Account to Automation Account
+.DESCRIPTION
+Module should be uploaded to Azure Storage Account first
+.PARAMETER automationAccountName
+Azure Automation Account name
+.PARAMETER subscriptionName
+Subscription name of the Azure Automation Account
+.PARAMETER resourceGroupName
+Resource group name of the Azure Automation Account
+.PARAMETER moduleName
+Name of the module
+.PARAMETER url
+URL of the the publicly accessible module.
+If URL is a private blob URL, the URL should be a SAS URL.
+.NOTES
+For some reason Set-AzureRmAutomationModule doesn't work. 
+New-AzureRmAutomationModule have to be used even if the module already exsists on Automation Accunt
+#>
+function UpdateMouduleToAutomationAccount ( 
+     [string] $automationAccountName
+    ,[string] $subscriptionName
+    ,[string] $resourceGroupName
+    ,[string] $moduleName
+    ,[string] $url ) {
+    
+    $subscriptionName, $automationAccountName, $resourceGroupName = DefaultIfNotSpecifiedAA $subscriptionName $automationAccountName $resourceGroupName
+    Write-Host "Adding module '$moduleName' to Azure Automation account..." -ForegroundColor Green
+    $null = New-AzureRmAutomationModule -AutomationAccountName $automationAccountName -ResourceGroupName $resourceGroupName -Name $moduleName -ContentLink $url -ErrorAction Stop
```

Comment:
Note that, instead of assigning the output to a variable, you can use 
```powershell
New-AzureRmAzutomationModule | Out-Null
```

Reply:
I use $null var for visibility.